File: spring-integration/pollable-channels/src/main/java/com/ddubson/PrintService.java
package com.ddubson;

import org.springframework.messaging.Message;
import org.springframework.messaging.support.MessageBuilder;

public class PrintService {
    public Message<?> print(Message<String> msg) {
        System.out.println("PrintService: "+ msg.getPayload());
        return MessageBuilder.withPayload("PrintService Response Message: " + msg.getHeaders().get("id")).build();
    }
}

File: spring-integration/subscribable-channels/channel-interceptors/src/main/kotlin/com/ddubson/integration/services/PrintService.kt
package com.ddubson.integration.services

import com.ddubson.ANSIColor
import com.ddubson.LogAdapter
import org.springframework.messaging.Message

class PrintService(val logAdapter: LogAdapter) {
    fun print(msg: Message<String>) {
        logAdapter.info(
                "[3] Consuming off of input channel -- [${Thread.currentThread().name}] ${msg.payload}"
                , ANSIColor.ANSI_CYAN)
    }
}